src/Microsoft.FeatureManagement/Targeting/ContextualTargetingFeatureVariantAssigner.cs

```diff
@@ -67,69 +67,128 @@ public ValueTask<FeatureVariant> AssignVariantAsync(FeatureVariantAssignmentCont
 
             FeatureVariant variant = null;
 
-            double cumulativePercentage = 0;
+            var lookup = new Dictionary<FeatureVariant, TargetingFilterSettings>();
```

Comment:
Will this dictionary be thread safe?

Reply:
`Dictionary` itself is not thread safe, but since this is a function local variable we don't have to worry about multi-threaded access.